spack/package.py

```diff
@@ -230,24 +235,7 @@ def setup_build_environment(self, env):
         env.set("COMP_SYS", comp_sys)
 
     def build(self, pkg, spec, prefix):
-        # Get source files for netCDF4 builds
-        if self.spec.satisfies("+netcdf4"):
```

Comment:
Will removing this block adversely affect installing older versions?

Reply:
Yeah, basically download the spack repo, swap in this recipe for the existing one, `. <spackdir>/share/spack/setup-env.sh`, then `spack install wgrib2@2.0.8`. I started to test it out just now and ran into an issue with the recipe logic on line 225 because the 'netcdf' variant doesn't exist for that version. When I worked around that, it failed on the netcdf step, right now I'm working on figuring out if it's related to these changes.

Reply:
It's building for me now. The two further issues I ran into aren't introduced by your PR, but FYI:

1. The `parallel = False` setting isn't getting respected, so it continues to fail on parallel builds. There might be some cleaner or more Spack-ish way to do it, but for now I can get it to work by changing the `make()` call to `make("-j1")`.
2. Older versions of wgrib2 have lines like this in `makefile`:
```Makefile
ifneq ($(USE_IPOLATES),0)
a:=$(shell echo "\#define IPOLATES_LIB \"`basename ${ipdir}`\"" >> ${CONFIG_H})
endif
a:=$(shell echo "\#define USE_IPOLATES ${USE_IPOLATES}" >> ${CONFIG_H})
```
where the escaped #'s don't behave the same way in more recents versions of gmake. I would suggest adding:
```python
    @when("@:2 ^gmake@:4.1")
    def patch(self):
        filter_file("\\\#define", "#define", "makefile")
```

Reply:
As far as the variants that only exist in one or the other of makefile vs. cmake builds, you might consider having `variant_map` contain the ones they have in common, then have separate `variant_map_cmake` and `variant_map_makefile` to which you could add `variant_map`, and use the corresponding one in CMakeBuilder and MakefileBuilder.

Reply:
Apologies, this one keeps falling under my radar. Does this recent update look right?